addOns/oast/src/main/java/org/zaproxy/addon/oast/OastApi.java

```diff
@@ -0,0 +1,198 @@
+                        ACTION_SET_CALLBACK_OPTIONS,
+                        new String[] {
+                            PARAM_SERVER, PARAM_LOCAL_ADDR, PARAM_REMOTE_ADDR, PARAM_PORT
+                        }));
+        this.addApiAction(
+                new ApiAction(
+                        ACTION_SET_INTERACTSH_OPTIONS,
+                        new String[] {PARAM_SERVER, PARAM_POLL_IN_SECS, PARAM_AUTH_TOKEN}));
+    }
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    @Override
+    public ApiResponse handleApiAction(String name, JSONObject params) throws ApiException {
+        switch (name) {
+            case ACTION_SET_ACTIVE_SERVICE:
+                try {
+                    ext.setActiveScanOastService(params.getString(PARAM_NAME));
+                } catch (IllegalArgumentException e) {
+                    throw new ApiException(
+                            ApiException.Type.ILLEGAL_PARAMETER, params.getString(PARAM_NAME));
+                }
+                break;
+            case ACTION_SET_BOAST_OPTIONS:
+                ext.getBoastService().getParam().setBoastUri(params.getString(PARAM_SERVER));
+                try {
+                    ext.getBoastService()
+                            .getParam()
+                            .setPollingFrequency(
+                                    Integer.parseInt(params.getString(PARAM_POLL_IN_SECS)));
+                } catch (NumberFormatException e) {
+                    throw new ApiException(
+                            ApiException.Type.ILLEGAL_PARAMETER,
+                            params.getString(PARAM_POLL_IN_SECS));
+                }
+                break;
+            case ACTION_SET_CALLBACK_OPTIONS:
+                ext.getCallbackService()
+                        .getParam()
+                        .setLocalAddress(params.getString(PARAM_LOCAL_ADDR));
+                ext.getCallbackService()
+                        .getParam()
+                        .setRemoteAddress(params.getString(PARAM_REMOTE_ADDR));
+                try {
+                    ext.getCallbackService().getParam().setPort(params.getInt(PARAM_PORT));
+                } catch (NumberFormatException e) {
+                    throw new ApiException(
+                            ApiException.Type.ILLEGAL_PARAMETER, params.getString(PARAM_PORT));
+                }
+                break;
+            case ACTION_SET_INTERACTSH_OPTIONS:
+                ext.getInteractshService().getParam().setServerUrl(params.getString(PARAM_SERVER));
+                ext.getInteractshService()
+                        .getParam()
+                        .setAuthToken(params.getString(PARAM_AUTH_TOKEN));
+                try {
+                    ext.getInteractshService()
+                            .getParam()
+                            .setPollingFrequency(
+                                    Integer.parseInt(params.getString(PARAM_POLL_IN_SECS)));
```

Comment:
params.getInt

Reply:
It's better to use e.g. `ApiUtils.getIntParam(params, PARAM_POLL_IN_SECS)` which already handles the possible missing/invalid values.